### PR TITLE
Merge State Cleanup

### PR DESCRIPTION
**Pull request by nisdas**
**What type of PR is this?**

Cleanup

**What does this PR do? Why is it needed?**

- [x] Cleans up some things from our merge related state. 

**Which issues(s) does this PR fix?**

N.A

**Other notes for review**

### CODE REVIEW

**Review by rkapka — DISMISSED** [review]

File: beacon-chain/cache/sync_committee_head_state.go [hunk 1]
```diff
@@ -51,13 +50,14 @@ func (c *SyncCommitteeHeadStateCache) Get(slot types.Slot) (state.BeaconState, e
...
-	st, ok := val.(*v2.BeaconState)
+	st, ok := val.(state.BeaconState)
 	if !ok {
-		st, ok = val.(*v3.BeaconState)
-		if !ok {
-			return nil, ErrIncorrectType
-		}
+		return nil, ErrIncorrectType
+	}
+	switch st.Version() {
```
Comment (terencechain) [inline beacon-chain/cache/sync_committee_head_state.go:57]: Much better than what I did. Thank you

**Review by terencechain — APPROVED** [review]

**Comment by prestonvanloon** [comment]
Can this be merged?

File: beacon-chain/cache/sync_committee_head_state.go [hunk 2]
```diff
@@ -51,13 +50,14 @@ func (c *SyncCommitteeHeadStateCache) Get(slot types.Slot) (state.BeaconState, e
...
+	case version.Altair, version.Merge:
+	default:
+		return nil, ErrIncorrectType
 	}
```
Comment (prestonvanloon) [inline beacon-chain/cache/sync_committee_head_state.go:61]: What is the purpose of this check? Only altair and merge versions work?

Comment (nisdas) [reply beacon-chain/cache/sync_committee_head_state.go:61]: Yeap, phase0 states do not have sync committees.

File: beacon-chain/state/v3/state_trie.go [hunk 3]
```diff
@@ -10,6 +10,7 @@ import (
 	"github.com/prysmaticlabs/prysm/beacon-chain/state/fieldtrie"
 	"github.com/prysmaticlabs/prysm/beacon-chain/state/stateutil"
 	"github.com/prysmaticlabs/prysm/beacon-chain/state/types"
+	"github.com/prysmaticlabs/prysm/config/features"
```
Comment (rkapka) [inline beacon-chain/state/v3/state_trie.go:13]: Please apply the same changes in `/state/native-state`

**Review by rkapka — CHANGES_REQUESTED** [review]

**Review by rauljordan — DISMISSED** [review]

**Review by rkapka — APPROVED** [review]

**Pull request merged**